--- maven-1.0.2/bin/maven~	2004-12-07 13:13:46.000000000 +0200
+++ maven-1.0.2/bin/maven	2007-04-20 00:03:14.620618599 +0300
@@ -51,13 +51,13 @@
 
 if [ -z "$MAVEN_HOME" ] ; then
   # try to find MAVEN
-  if [ -d /opt/maven ] ; then
+  if [ -d /usr/share/maven ] ; then 
+    MAVEN_HOME=/usr/share/maven
+  elif [ -d /opt/maven ] ; then
     MAVEN_HOME=/opt/maven
-  fi
-
-  if [ -d "$HOME/maven" ] ; then
+  elif [ -d "$HOME/maven" ] ; then
     MAVEN_HOME="$HOME/maven"
-  fi
+  else
 
   ## resolve links - $0 may be a link to maven's home
   PRG=$0
@@ -83,6 +83,7 @@
   MAVEN_HOME=`cd "$MAVEN_HOME" && pwd`
 
   cd "$saveddir"
+  fi
 fi
 
 # For Cygwin, ensure paths are in UNIX format before anything is touched
@@ -97,6 +98,10 @@
     CLASSPATH=`cygpath --path --unix "$CLASSPATH"`
 fi
 
+if [ -z "$JAVA_HOME" -a -x /usr/lib/java/bin/java ]; then
+  JAVA_HOME=/usr/lib/java
+fi
+
 if [ -z "$JAVACMD" ] ; then
   if [ -n "$JAVA_HOME"  ] ; then
     if [ -x "$JAVA_HOME/jre/sh/java" ] ; then
