diff -up ./maven2/maven-core/src/main/java/org/apache/maven/cli/MavenCli.java.sav ./maven2/maven-core/src/main/java/org/apache/maven/cli/MavenCli.java
--- ./maven2/maven-core/src/main/java/org/apache/maven/cli/MavenCli.java.sav	2009-01-29 16:47:45.000000000 -0500
+++ ./maven2/maven-core/src/main/java/org/apache/maven/cli/MavenCli.java	2009-01-29 16:48:00.000000000 -0500
@@ -645,7 +645,7 @@ public class MavenCli
                 "Force the use of an alternate POM file." ).create( ALTERNATE_POM_FILE ) );
 
             options.addOption(
-                OptionBuilder.withLongOpt( "define" ).hasArg().withDescription( "Define a system property" ).create(
+                OptionBuilder.withLongOpt( "define" ).hasArgs(1).withDescription( "Define a system property" ).create(
                     SET_SYSTEM_PROPERTY ) );
             options.addOption(
                 OptionBuilder.withLongOpt( "offline" ).withDescription( "Work offline" ).create( OFFLINE ) );
